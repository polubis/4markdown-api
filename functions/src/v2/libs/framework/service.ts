import {
  DocumentData,
  DocumentReference,
  DocumentSnapshot,
} from 'firebase-admin/firestore';

type ServiceResult<TModel> = Promise<{
  data: TModel | undefined;
  ref: DocumentReference<DocumentData>;
}>;

type Service<TPayload extends Record<string, any>, TModel> = (
  payload: TPayload & {
    action?: (
      ref: DocumentReference<DocumentData>,
    ) => Promise<DocumentSnapshot<DocumentData>>;
  },
) => ServiceResult<TModel>;

export type { Service };
