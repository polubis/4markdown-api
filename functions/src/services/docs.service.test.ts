import { DocsService } from './docs.service';

describe(`Docs updating works when: `, () => {
  it(`throws an error when invalid name`, async () => {
    try {
      await DocsService.update(`3123-dsad-323`, {
        name: `invalid-name`,
        code: `dasdad`,
        visibility: `public`,
        id: `312313`,
      });
    } catch (error) {
      expect((error as Error).message).toBe(`Wrong name format`);
    }
  });
});
