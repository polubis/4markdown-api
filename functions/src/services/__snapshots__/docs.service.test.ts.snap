// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Docs updating works when:  throws an error if docs collection is not created yet 1`] = `"{"symbol":"not-found","content":"Not found","message":"Not found"}"`;

exports[`Docs updating works when:  throws an error when invalid name 1`] = `"{"symbol":"bad-request","content":"Wrong name format","message":"Wrong name format"}"`;
